pkg/provisioner/nodeprovisioner.go

```diff
@@ -184,3 +221,66 @@ func isCorruptedMount(target string) bool {
 	fmt.Printf("IsCorruptedDir(%s) returned with error: %v", target, pathErr)
 	return pathErr != nil && mount.IsCorruptedMnt(pathErr)
 }
+
+func (p *NodeProvisioner) MarkRecovery(diskURI string) {
+	if p.deviceChecker == nil {
+		deviceChecker := newDeviceCheckerEntry(diskURI)
+		p.deviceChecker = &deviceChecker
+	} else {
+		p.deviceChecker.lock.Lock()
+		defer p.deviceChecker.lock.Unlock()
+		p.deviceChecker.curRetry++
+	}
+}
+
+// Given the amount of time, it might take for iscsi scan to complete, allow retries for getDiskPathWithLun operation
+// In case of persisting failure, we will back off the max retry count
+func (p *NodeProvisioner) MountRecoveryRequired(diskURI string) bool {
+	p.deviceChecker.lock.Lock()
+	defer p.deviceChecker.lock.Unlock()
+	if p.deviceChecker.diskURI != diskURI {
+		return false
+	}
+	defer func() { p.deviceChecker.curRetry++ }()
+	if p.deviceChecker.curRetry >= p.deviceChecker.maxRetry {
+		p.deviceChecker.detachInProcess = true
+		return true
+	}
+	return false
+}
+
+func (p *NodeProvisioner) IsDetachInProcess(diskURI string) bool {
+	p.deviceChecker.lock.Lock()
```

Comment:
There's no guarantee that `deviceChecker` has been allocated at this point, so you may get a `nil` pointer error.